src/Button/GeoLocationButton/GeoLocationButton.jsx

```diff
@@ -0,0 +1,265 @@
+     * @type {Boolean}
+     */
+    showMarker: PropTypes.bool,
+
+    /**
+     * Whether to follow the current position.
+     *
+     * @type {Boolean}
+     */
+    follow: PropTypes.bool
+  };
+
+  /**
+   * The default properties.
+   * @type {Object}
+   */
+  static defaultProps = {
+    onGeolocationChange: () => undefined,
+    onError: () => undefined,
+    showMarker: true,
+    follow: true
+  }
+
+  /**
+   * Creates the MeasureButton.
+   *
+   * @constructs MeasureButton
+   */
+  constructor(props) {
+    super(props);
+
+    this.positions = new OlGeomLineString([], 'XYZM');
+
+    this.state = {
+    };
+  }
+
+  onGeolocationChange = () => {
+    const position = this.geolocationInteraction.getPosition();
+    const accuracy = this.geolocationInteraction.getAccuracy();
+    let heading = this.geolocationInteraction.getHeading() || 0;
+    const speed = this.geolocationInteraction.getSpeed() || 0;
+
+    const x = position[0];
+    const y = position[1];
+    const fCoords = this.positions.getCoordinates();
+    const previous = fCoords[fCoords.length - 1];
+    const prevHeading = previous && previous[2];
+    if (prevHeading) {
+      let headingDiff = heading - MathUtil.mod(prevHeading);
+
+      // force the rotation change to be less than 180°
+      if (Math.abs(headingDiff) > Math.PI) {
+        var sign = (headingDiff >= 0) ? 1 : -1;
+        headingDiff = -sign * (2 * Math.PI - Math.abs(headingDiff));
+      }
+      heading = prevHeading + headingDiff;
+    }
+    this.positions.appendCoordinate([x, y, heading, Date.now()]);
+
+    // only keep the 20 last coordinates
+    this.positions.setCoordinates(this.positions.getCoordinates().slice(-20));
+
+    if (this.markerEl) {
+      if (heading && speed) {
+        this.markerEl.src = mapMarkerHeading;
+      } else {
+        this.markerEl.src = mapMarker;
+      }
+    }
+
+    this.updateView();
+
+    this.props.onGeolocationChange({
+      position,
+      accuracy,
+      heading,
+      speed
+    });
+  }
+
+  onGeolocationError = () => {
+    this.props.onError();
+  }
+
+  /**
+   * Called when the button is toggled, this method ensures that everything
+   * is cleaned up when unpressed, and that geolocating can start when pressed.
+   *
+   * @method
+   */
+  onToggle = (pressed) => {
+    if (!pressed && this.geolocationInteraction) {
+      this.geolocationInteraction.un('change', this.onGeolocationChange);
+      this.geolocationInteraction = null;
+      if (this.marker) {
+        this.props.map.removeOverlay(this.marker);
+        this.markerEl.parentNode.removeChild(this.markerEl);
+      }
+    }
+    if (!pressed) {
+      return;
+    }
+    const map = this.props.map;
+    const view = map.getView();
+
+    // Geolocation Control
+    this.geolocationInteraction = new OlGeolocation({
+      projection: view.getProjection(),
+      trackingOptions: {
+        maximumAge: 10000,
+        enableHighAccuracy: true,
+        timeout: 600000
+      }
```

Comment:
`trackingOptions` should be configurable via a `prop`.

Reply:
Done